tests/components/medcom_ble/test_config_flow.py

```diff
@@ -0,0 +1,230 @@
+"""Test the Medcom Inspector BLE config flow."""
+from unittest.mock import patch
+
+from bleak import BleakError
+from medcom_ble import MedcomBleDevice
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.medcom_ble.const import DOMAIN
+from homeassistant.const import CONF_ADDRESS
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from . import (
+    MEDCOM_DEVICE_INFO,
+    MEDCOM_SERVICE_INFO,
+    UNKNOWN_SERVICE_INFO,
+    patch_async_ble_device_from_address,
+    patch_async_setup_entry,
+    patch_medcom_ble,
+)
+
+from tests.common import MockConfigEntry
+
+
+@pytest.mark.parametrize(
+    ("medcom_service", "ble_device"),
+    [
+        (
+            MEDCOM_SERVICE_INFO,
+            MedcomBleDevice(
+                manufacturer="International Medcom",
+                model="Inspector BLE",
+                model_raw="Inspector-BLE",
+                name="Inspector BLE",
+                identifier="a0d95a570b00",
+            ),
+        )
+    ],
+)
+async def test_bluetooth_discovery(
+    hass: HomeAssistant, medcom_service, ble_device
+) -> None:
+    """Test discovery via bluetooth with a valid device."""
+    with patch_async_ble_device_from_address(medcom_service), patch_medcom_ble(
+        ble_device
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_BLUETOOTH},
+            data=MEDCOM_SERVICE_INFO,
+        )
+
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "bluetooth_confirm"
+    assert result["description_placeholders"] == {
+        "name": "Inspector BLE (a0d95a570b00)"
+    }
+
+    with patch_async_setup_entry():
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input={"not": "empty"}
+        )
+    await hass.async_block_till_done()
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["title"] == "Inspector BLE (a0d95a570b00)"
+    assert result["result"].unique_id == "cc:cc:cc:cc:cc:cc"
+
+
+async def test_bluetooth_discovery_no_BLEDevice(hass: HomeAssistant) -> None:
+    """Test discovery via bluetooth but there's no BLEDevice."""
+    with patch_async_ble_device_from_address(None):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_BLUETOOTH},
+            data=MEDCOM_SERVICE_INFO,
+        )
+    assert result["type"] == FlowResultType.ABORT
+    assert result["reason"] == "cannot_connect"
+
+
+@pytest.mark.parametrize(
+    ("medcom_service", "exc", "reason"),
+    [
+        (MEDCOM_SERVICE_INFO, Exception(), "unknown"),
+        (MEDCOM_SERVICE_INFO, BleakError(), "cannot_connect"),
+    ],
+)
```

Comment:
Is this better as far as using `@pytest.mark.parametrize` is concerned ?

Reply:
Yes, but you don't need to `parametrize` `MEDCOM_SERVICE_INFO` since its always the same